tests/Unit/ValidatorTest.php

```diff
@@ -0,0 +1,130 @@
+<?php
+/**
+ * Unit Tests: Validator functionality
+ *
+ * @package Parsely\Tests
+ */
+
+declare(strict_types=1);
+
+namespace Parsely\Tests\Unit;
+
+use Parsely\Validator;
+use Yoast\WPTestUtils\BrainMonkey\TestCase;
+
+
+/**
+ * Unit tests for the functions of the Validator class.
+ *
+ * @since 3.3.0
+ * @since 3.9.0 Renamed/moved to ValidatorTest.php.
+ */
+final class ValidatorTest extends TestCase {
+	/**
+	 * Verifies that valid Site IDs pass validation.
+	 *
+	 * @since 3.3.0
+	 * @since 3.9.0 Renamed to test_validate_valid_site_ids().
+	 *
+	 * @covers \Parsely\Validator::validate_site_id
+	 */
+	public function test_validate_valid_site_ids(): void {
+		$valid_site_ids = array(
+			'test.com',
+			'www.test.com',
+			'subdomain.test.com',
+			'www.subdomain.test.com',
+			'subdomain.subdomain.test.com',
+		);
+
+		foreach ( $valid_site_ids as $valid_site_id ) {
+			self::assertTrue( Validator::validate_site_id( $valid_site_id ) );
+		}
+	}
+
+	/**
+	 * Verifies that invalid Site IDs fail validation.
+	 *
+	 * @since 3.3.0
+	 * @since 3.9.0 Extracted from test_validate_site_ids() as a separate test.
+	 *
+	 * @covers \Parsely\Validator::validate_site_id
+	 */
+	public function test_validate_invalid_site_ids(): void {
+		$invalid_values = array(
```

Comment:
Data Providers are a nice way to loop through test cases: https://phpunit.de/manual/3.7/en/writing-tests-for-phpunit.html#writing-tests-for-phpunit.data-providers.examples.DataTest.php

Reply:
Thanks for the mention. I discovered data providers very recently and I'm using them in one of the new PRs. In the specific case, I like the separation because the function names refer to what is exactly being tested. Obviously, this is subjective.